elasticdl/python/tests/worker_ps_interaction_test.py

```diff
@@ -78,18 +92,96 @@ def _create_pserver_and_channel(self, ports):
                     ),
                 ],
             )
-            channels.append(channel)
-        return pservers, channels
-
-    def _restart_pserver(self):
-        # Stop first
-        self.tearDown()
-        # Start again
-        self._pserver, self._channel = self._create_pserver_and_channel(
-            self._ports
-        )
+            self._channels.append(channel)
+
+        self._pservers = []
+        self._workers = []
+
+    def tearDown(self):
+        for pserver in self._pservers:
+            pserver.server.stop(0)
+
+    def _create_pserver(self, model_def):
```

Comment:
We can add `num_ps` argument in this method, and create an arbitrary number of ps. This will make future unittests with different number of ps/worker much easier.

Reply:
Done